File: near-library/src/main/java/com/example/near_library/model/Endpoint.java
package com.example.near_library.model;

import android.support.annotation.NonNull;

/** Represents a device we can talk to. */
public class Endpoint {

    @NonNull
    private final String id;

    @NonNull
    private final String name;

    public Endpoint(@NonNull String id, @NonNull String name) {
        this.id = id;
        this.name = name;
    }

    @NonNull
    public String getId() {
        return id;
    }

    @NonNull
    public String getName() {
        return name;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj != null && obj instanceof Endpoint) {
            Endpoint other = (Endpoint) obj;
            return id.equals(other.id);
        }
        return false;
    }

    @Override
    public int hashCode() {
        return id.hashCode();
    }

    @Override
    public String toString() {
        return String.format("Endpoint{id=%s, name=%s}", id, name);
    }

    public String encode() {
        return String.format("%s;%s", id, name);
    }

    public static Endpoint decode(String encodingStr) {
        String[] components = encodingStr.split(";");
        return new Endpoint(components[0], components[1]);
    }
}

File: app/src/main/java/com/mtoader/near/model/Endpoint.kt
package com.mtoader.near.model

/** Represents a device we can talk to.  */
class Endpoint constructor(val id: String, val name: String) {

    override fun equals(obj: Any?): Boolean {
        if (obj != null && obj is Endpoint) {
            val other = obj as Endpoint?
            return id == other!!.id
        }
        return false
    }

    override fun hashCode(): Int {
        return id.hashCode()
    }

    override fun toString(): String {
        return String.format("Endpoint{id=%s, name=%s}", id, name)
    }
}